python/src/spark_rapids_ml/umap.py

```diff
@@ -675,14 +698,14 @@ def _construct_umap() -> CumlT:
             from .utils import cudf_to_cuml_array
 
             embedding = (
-                self.embedding_
-                if isinstance(self.embedding_, np.ndarray)
-                else self.embedding_.value
+                self.embedding_[0].value
+                if len(self.embedding_) == 1
+                else np.concatenate([chunk.value for chunk in self.embedding_])
```

Comment:
Sorry for the late review, Please don't use any "self" in the function running on the executor side, which will pack everything of "self" and send to executor from driver.